### PR TITLE
Ensuring that multiple titles can be updated for an approved Work

### LINKED ISSUE #833: Only curators should be able to modify approved items
Curators should be able to modify objects after they have been approved.

Test to ensure that curators can modify objects that they themselves submitted after they have been approved.

Additional titles for a Work are not being saved *when the work has already been approved*. Although I can edit the work and add a new title via the `+` button, the title is **not** saved on the work.
 
add_title_edit

No error is reported to the user during Save, but the system goes back to the home page rather than displaying the Show page for the work after save.

The system works as expected when the title is added to a work that has not yet been approved.

Related to https://github.com/pulibrary/pdc_discovery/issues/340

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #833

### CODE REVIEW

File: spec/requests/works_spec.rb [hunk 1]
```diff
@@ -53,6 +53,99 @@
...
+              expect(titles.length).to eq(2)
+
+              subtitle = titles.last
+              expect(subtitle).to be_a(Hash)
+              expect(subtitle).to include("title" => "the subtitle")
+              expect(subtitle).to include("title_type" => "Subtitle")
+            end
+          end
+
+          context "when authenticated as the submitter of the Work" do
```
Comment (jrgriffiniii) [inline spec/requests/works_spec.rb:108]: This seems to be the failing case. I can ensure that this is fully resolved.

**Comment by jrgriffiniii** [comment]
I was failing to reproduce these errors locally, I am going to proceed with resolving the outstanding failing test.

File: app/controllers/works_controller.rb [hunk 2]
```diff
@@ -114,8 +114,6 @@ def update
     if current_user.blank? || !@work.editable_by?(current_user)
       Rails.logger.warn("Unauthorized attempt to update work #{@work.id} by user #{current_user.uid}")
       redirect_to root_path
-    elsif @work.approved? && @work.submitted_by?(current_user)
```
Comment (carolyncole) [inline app/controllers/works_controller.rb:117]: The requirement that a user can not edit an item they deposited was ticketed here: https://github.com/pulibrary/pdc_describe/issues/255.  Does removing this code introduce a bug?

**Review by carolyncole — CHANGES_REQUESTED** [review]
@jrgriffiniii I am unsure if by fixing this "bug" we are introducing a bug.  We should check to see if we still want to limit when a depositor can edit the metadata before this gets merged.

I have no technical issue with this PR and I am not really requesting any changes,  I just want to block it from being merged until we can discuss it with RDOS.

**Comment by jrgriffiniii** [comment]
Given the discussion held on 01/20/23, it was confirmed by the Product Owners that submitters should *not* be able to modify metadata.

**Pull request closed without merging**